### DISCUSSION TITLE
Installing shadcn/ui on Next.js 15 RC

### DISCUSSION THREAD

**Discussion by tjhoo** [shadcn-ui/ui#3988] [General]
Hi,
I encouter the following issue when installing shadcn/ui in a Next.js 15 RC.
```
$ npx create-next-app@rc --empty dummy
√ Would you like to use TypeScript? ... No / Yes
√ Would you like to use ESLint? ... No / Yes
√ Would you like to use Tailwind CSS? ... No / Yes
√ Would you like to use `src/` directory? ... No / Yes
√ Would you like to use App Router? (recommended) ... No / Yes
√ Would you like to use Turbopack for next dev? ... No / Yes
√ Would you like to customize the default import alias (@/*)? ... No / Yes
...
$ npx shadcn-ui@latest init
√ Which style would you like to use? » Default
√ Which color would you like to use as base color? » Slate
√ Would you like to use CSS variables for colors? ... no / yes
...
Command failed with exit code 1: npm install tailwindcss-animate class-variance-authority clsx tailwind-merge lucide-react
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR!
npm ERR! While resolving: dummy@0.1.0
npm ERR! Found: react@19.0.0-rc-f994737d14-20240522
npm ERR! node_modules/react
npm ERR!   react@"19.0.0-rc-f994737d14-20240522" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer react@"^16.5.1 || ^17.0.0 || ^18.0.0" from lucide-react@0.394.0
npm ERR! node_modules/lucide-react
npm ERR!   lucide-react@"*" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR!
...
```
How can we specify `--force` option when installing `shadcn/ui`? Or is there a better way to workaround this?

**Comment by rogueturnip** [comment]
Has anyone found a way to run --force or --legacy-peer-deps on the init?

**Comment by yahyazgour** [comment]
Hello @tjhoo @rogueturnip 

I was able to make it work on rc using either `npx --force shadcn-ui@latest init` or `npx --legacy-peer-deps shadcn-ui@latest init`

You might also have to apply the same thing to:
1 - some **Shadcn** components or **v0.dev** components like `npx --legacy-peer-deps v0 add 5NuVtmUwLid`.
2 - some **@radix-ui** packages, had the issue with **react-dialog** and fixed it through `npm --legacy-peer-deps i @radix-ui/react-dialog`

**Reply by jakubiwaszkiewicz** [reply]
Thanks! In my case I had to run `npx --legacy-peer-deps shadcn@latest init -d` instead of `npx --legacy-peer-deps shadcn-ui@latest init`

**Reply by atulbhatt-system32** [reply]
Yeah I had the same experience:
https://youtu.be/8NiSPKK29ig

**Reply by atulbhatt-system32** [reply]
Now shadcn also prompts for this before installing.

<img width="955" alt="Screenshot 2024-11-04 at 8 28 51 PM" src="https://github.com/user-attachments/assets/c6b760ad-6826-4ed4-b847-bb7fbc12ab5e">

**Comment by atulbhatt-system32** [comment]
https://youtu.be/8NiSPKK29ig

here's the video on the same. Just in case ;)

**Comment by kirankumarhexafoldtech** [comment]
npx --legacy-peer-deps shadcn@latest init will works fine

**Reply by BrunoPierca** [reply]
Just in case, this same command **will** work when adding components too

**Comment by tutorialsmaterials** [comment]
check the blog solve this problem 
https://tutorialsmaterial.com/post/ShadCn-Nextjs-15

**Reply by k1rana** [reply]
That blog posts seem to be written without intention

**Comment by pasternak-karmel** [comment]
You have just to install this npm install tailwindcss-animate class-variance-authority clsx tailwind-merge before doing npx shadcn@latest init

**Reply by Ngumi22** [reply]
This worked well for me. I ran this first npm install tailwindcss-animate class-variance-authority clsx tailwind-merge and then npx shadcn@latest init and no errors at all

**Reply by atulbhatt-system32** [reply]
@radix-ui/react-icons@1.3.1 fails at here

**Reply by Ngumi22** [reply]
I realised you have to add this also every time --legacy-peer-deps when installing all packages so I went back to Next 14 for now until everything is okay

**Reply by atulbhatt-system32** [reply]
Yes I have  been doing this : https://youtu.be/8NiSPKK29ig

**Reply by Ngumi22** [reply]
That seems to be the only solution at the moment.

**Comment by donghoon-song** [comment]
It provides migration guide :)
https://x.com/shadcn/status/1851323295293194453
